CIP-0XXX/README.md

```diff
@@ -0,0 +1,561 @@
+2. The server either accepts the message (and returns a confirmation) or rejects it (and returns the reason)
+
+### State machine
+
+| Agency ||
+|--------|------| 
+| Client has Agency | StIdle | 
+| Server has Agency | StBusy |
+
+```mermaid
+stateDiagram-v2   
+
+  classDef White fill:white,stroke:white
+  classDef Black fill:white,stroke:black
+  classDef Blue fill:white,stroke:blue
+  classDef Green fill:white,stroke:green
+
+  start:::White --> StIdle:::Blue;
+  StIdle:::Blue --> StBusy:::Green : MsgNextMessage
+  StBusy:::Green --> StIdle:::Blue : MsgHasMessage
+  StBusy:::Green --> StIdle:::Blue : MsgTimeoutMessage
+  StIdle:::Blue --> StDone:::Black : MsgDone
+   
+```
+
+#### Protocol messages
+
+* **MsgNextMessage**: The client asks for the next message.
+* **MsgHasMessage(message)**: The server has received a message.
+* **MsgTimeoutMessage**: The server has not received a message and times out.
+* **MsgDone**: The client terminates the mini-protocol.
+
+#### Transition table
+
+| From state | Message | Parameters | To State|
+|--------|------|----|----| 
+| StIdle | MsgNextMessage |  | StBusy |
+| StBusy | MsgHasMessage | message | StIdle |
+| StBusy | MsgTimeoutMessage | reason | StIdle |
+| StIdle | MsgDone |  | StDone |
+
+# Rationale
+
+## Why are we proposing this CIP?
+
+### For Mithril
+
+* Mithril requires strong network foundations to support interactions between its various nodes:
+
+  * Mithril needs to exist in a decentralized context where multiple aggregators can operate seamlessly and independantly.
+  * Mithril needs participation of all or nearly all of the Cardano network SPOs to provide maximal security to the multi-signatures embedded in the certificates.
+  * Creating a separate network would entail significant costs and efforts (there are more than 3,000 SPOs which would need to be connected with resilient and secure network, and much more passive nodes).
+  * The Cardano network is very efficient for diffusion (e.g. broadcasting) which is precisely what is needed for Mithril.
+  * Mithril signer node needs to run on the same machine as the Cardano block producing node (to access the KES keys). It makes sense to use the same network layer, which will also facilitate a high level of participation.
+
+### For Leios
+
+> [!NOTE]
+> This section needs to be completed.
+
+### For Peras
+
+> [!NOTE]
+> This section needs to be completed.
+
+### For Cardano
+
+* Why would it be great for Cardano to support a decentralized message queue with its network?
+
+  * This is a required feature to make the Cardano ecosystem scalable.
+  * The design is versatile enough to support present and future use cases.
+
+# Information Diffusion Architecture
+
+In this section, we propose an architecture for `Cardano` and `Mithril`. Note,
+in this section, `Mithril` is used as a placeholder for a possible `Mithril`
+application: we've seen many ideas about how `Mithril` can be used in `Cardano`
+and all of them can follow this design.
```

Comment:
This is a weird section and does not fit as a top-level section the CIP template.

I think it's great that we can give more details how mithril and cardano processes would interact, but ultimately the CIP is about the network protocol.

Maybe it would fit better as an Appendix titled "Cardano/Mithril node architecture" or so?

Reply:
I disagree I think this should be clear from the beginning and it is not. In the abstract, motivation and protocol implementation sections, there's only mention of cardano-node and its network and it leads the reader to think this is going to be an addition to them. Sure mithril is about the protocol, but as we already have discussed there's no strong reason why this protocol should be added to the protocol stack of cardano-node. The viable option that we settled on was what this "architechure" section describes. This is the optimal route, even for Peras and Leios.